portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -47,23 +47,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String logoutUrl = userService.createLogoutURL(REDIRECT_URL_AFTER_LOGOUT);
-      saveComment(request.getParameter("display-name"), request.getParameter("comment"));
+      // TODO: Use nickname instead
+      saveComment(userService.getCurrentUser().getEmail(), request.getParameter("comment"));
       response.sendRedirect("/aboutme.html");
-    } else {
-      String loginUrl = userService.createLoginURL(REDIRECT_URL_AFTER_LOGIN);
-      response.sendRedirect(loginUrl);
     }
-
   }
 
   private String convertCommentsToJson() {
     Query query = new Query("Comment");
     PreparedQuery results = datastore.prepare(query);
+
```

Comment:
You should look into using gson. Less error prone. You can create a data class with displayName & comment as properties. You can do this in a follow up PR.